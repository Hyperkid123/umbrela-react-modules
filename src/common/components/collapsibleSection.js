import React, {Component} from 'react';
import PropTypes from 'prop-types';
import ExpansionPanel, {
  ExpansionPanelSummary,
  ExpansionPanelDetails,
} from 'material-ui/ExpansionPanel';
import Typography from 'material-ui/Typography';
import ExpandMoreIcon from 'material-ui-icons/ExpandMore';

export default class CollapsibleSection extends Component {
  render() {
    return (
      <ExpansionPanel>
        <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
          <Typography>Expansion Panel 1</Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails>
          <Typography>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex,
            sit amet blandit leo lobortis eget.
          </Typography>
        </ExpansionPanelDetails>
      </ExpansionPanel>
    );
  }
}

CollapsibleSection.propTypes = {
  sectionTitle: PropTypes.string.isRequired,
  children: PropTypes.object,
};
