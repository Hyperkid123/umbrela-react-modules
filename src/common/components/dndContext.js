import {DragDropContext} from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import React,{Component} from 'react';

export class DNDContext extends Component {
    render() {
        return this.props.children
    }
}



export default DragDropContext(HTML5Backend)(DNDContext);
