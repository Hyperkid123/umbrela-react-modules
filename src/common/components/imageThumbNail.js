import React from 'react';
import {ThumbNail} from '../styledComponents/containers';

export const ImageThumbNail = ({src, alt}) => (
  <div style={{margin: 5, display: 'inline-block'}}>
    <ThumbNail src={src} alt={alt}/>
  </div>
)
