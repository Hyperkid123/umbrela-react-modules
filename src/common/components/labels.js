import React from 'react';
import {
  LabelContainer,
} from '../styledComponents/containers';

export const TextFieldComent = ({label, alignRight, error}) => (
  <LabelContainer alignRight error={error}>
    {label}
  </LabelContainer>
);
