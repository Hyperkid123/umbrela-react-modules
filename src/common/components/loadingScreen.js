import React from 'react';
import { CircularProgress } from 'material-ui/Progress';
import {Flex} from '../styledComponents/containers';

const LoadingScreen = () => (
    <Flex horizontalCenter grow verticalCenter>
      <CircularProgress size={80} thickness={5} />
    </Flex>
);

export default LoadingScreen;
