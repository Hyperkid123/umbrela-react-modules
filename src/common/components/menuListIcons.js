import React from 'react';
import ExpandLess from 'material-ui-icons/ExpandLess';
import ExpandMore from 'material-ui-icons/ExpandMore';
import IconButton from 'material-ui/IconButton';
import {
  MenuListIconsContainer,
  MenuItem,
  MenuListLabel,
  Flex,
} from '../styledComponents/containers';
import {DefaultLink} from '../styledComponents/typography';
import Tooltip from 'material-ui/Tooltip';

export const ChevronUp = ({tooltip}) => (
  <IconButton tooltip={tooltip}>
    <ExpandLess />
  </IconButton>
);

export const ChevronDown = ({tooltip}) => (
  <IconButton tooltip={tooltip}>
    <ExpandMore />
  </IconButton>
);

export const MenuListIconsWrapper = ({tooltipUp, tooltipDown}) => (
  <MenuListIconsContainer>
    <ChevronUp tooltip={tooltipUp}/>
    <ChevronDown tooltip={tooltipDown}/>
  </MenuListIconsContainer>
)

export const MenuListItem = ({link, label, onClick, sheetId, active, tooltipLabel, dragging, preventDrag}) => (
        <Tooltip
          title={tooltipLabel}
          enterDelay={500}
          disableTriggerFocus={dragging || preventDrag}
          disableTriggerHover={dragging || preventDrag}
          disableTriggerTouch={dragging || preventDrag}
          placement='right-end'
        >
          <MenuItem dragging={dragging} data-tip={tooltipLabel} active={active} onClick={() => onClick(sheetId)}>
          <Flex verticalCenter>
          <MenuListLabel>
            <DefaultLink href={link}>{label}</DefaultLink>
          </MenuListLabel>
        </Flex>
    </MenuItem>
  </Tooltip>
)
