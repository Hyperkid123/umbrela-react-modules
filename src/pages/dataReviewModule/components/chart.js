import React,{Component} from 'react';
//import CloseQuestions from './CloseQuestions';
import OpenQuestion from './openQuestion';
//import MatrixQuestion from './MatrixQuestion';
import PropTypes from 'prop-types';
import {Flex} from '../../../common/styledComponents/containers';
//import LegendTable from './LegendTable';
//import DivideQuestion from './DivideQuestion';

import {
  OPEN_QUESTION,
  CLOSE_QUESTION,
  CLOSE_MULTI_QUESTION,
  CLOSE_WITH_OPEN_QUESTION,
  CLOSE_MULTI_WITH_OPEN_QUESTION,
  MATRIX_MULTI_IMAGE_QUESTION,
  MATRIX_MULTI_QUESTION,
  MATRIX_SINGLE_IMAGE_QUESTION,
  MATRIX_SINGLE_QUESTION,
  DIVIDE_QUESTION,
  ORDER_QUESTION,
  OPEN_WITH_IMAGE_QUESTIONS,
  IMAGE_OPTIONS_QUESTION,
  IMAGE_OPTIONS_MULTI_QUESTION,
  IMAGE_OPTIONS_WITH_OPEN_QUESTION,
  IMAGE_OPTIONS_MULTI_WITH_OPEN_QUESTION,
  MEDIA_QUESTION,
  MEDIA_MULTI_QUESTION,
  MEDIA_WITH_OPEN_QUESTION,
  MEDIA_MULTI_WITH_OPEN_QUESTION,
  isOptionImage,
  hasImagePreview,
} from '../../../common/chartTypes';

export default class Chart extends Component {

  renderChart = () => {
    switch (this.props.questionType) {
      case OPEN_QUESTION:
      case OPEN_WITH_IMAGE_QUESTIONS:
        return (
          <Flex>
            <OpenQuestion question={this.props.question} data={this.props.data}/>
          </Flex>
        );
      case CLOSE_QUESTION:
      case CLOSE_MULTI_QUESTION:
      case CLOSE_WITH_OPEN_QUESTION:
      case CLOSE_MULTI_WITH_OPEN_QUESTION:
      case IMAGE_OPTIONS_QUESTION:
      case IMAGE_OPTIONS_MULTI_QUESTION:
      case IMAGE_OPTIONS_WITH_OPEN_QUESTION:
      case IMAGE_OPTIONS_MULTI_WITH_OPEN_QUESTION:
      case MEDIA_QUESTION:
      case MEDIA_MULTI_QUESTION:
      case MEDIA_WITH_OPEN_QUESTION:
      case MEDIA_MULTI_WITH_OPEN_QUESTION:
        return (
            <Flex>
              {/**<CloseQuestions
                hasImagePreview={hasImagePreview(this.props.question.type)}
                showLegend={this.props.showLegend}
                chartType={this.props.chartType}
                data={this.props.data.barData}
                url={this.props.question.url}
              />
              <LegendTable isOptionImage={isOptionImage(this.props.question.type)} data={this.props.data}/>
            */}
            </Flex>
        );
      case  MATRIX_MULTI_IMAGE_QUESTION:
      case  MATRIX_MULTI_QUESTION:
      case  MATRIX_SINGLE_IMAGE_QUESTION:
      case  MATRIX_SINGLE_QUESTION:
        return (
          <Flex>
            {/**

            <MatrixQuestion
              data={this.props.data}
              chartType={this.props.chartType}
              showLegend={this.props.showLegend}
              question={this.props.question}
              isOptionImage={isOptionImage(this.props.question.type)}
            />
          */}
          </Flex>
        );
      case DIVIDE_QUESTION:
      case ORDER_QUESTION:
        return(
          {/**<DivideQuestion data={this.props.data}/>*/}
        );
      default:
        return null;

    }
  }

  render() {
    return this.renderChart();
  }
}

Chart.propTypes = {
  data: PropTypes.object.isRequired,
  chartType: PropTypes.string.isRequired,
  showLegend: PropTypes.bool.isRequired,
  questionType: PropTypes.string.isRequired,
  question: PropTypes.object.isRequired,
};
