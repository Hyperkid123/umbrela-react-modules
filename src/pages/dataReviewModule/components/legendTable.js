import React,{Component} from 'react';
import PropTypes from 'prop-types';
import Dialog from 'material-ui/Dialog';
import FiberManualRecord from 'material-ui-icons/FiberManualRecord'
import Table, { TableBody, TableCell, TableHead, TableRow, TableFooter } from 'material-ui/Table';
import {Flex} from '../../../common/styledComponents/containers';
import { localize } from 'react-localize-redux';

class LegendTable extends Component {

  constructor(props){
    super(props);
    this.state = {
      image: '',
      showImage: false,
    };
  }

  handleShowImage = (url) => {
    if(this.props.isOptionImage){
      this.setState({showImage: true, image: url});
    }
  }
  handleCloseImage = () => this.setState({showImage: false});

  renderTableBody = () => {
    const {labels, datasets} = this.props.data.barData.data;
    const {backgroundColor, data} = datasets[0];
    return labels.map((label, index) => {
      let buttonLabel = label;
      if(this.props.isOptionImage){
        buttonLabel = label.substr(label.lastIndexOf('/') + 1);
      }
      return (
        <TableRow key={index}
          data-tip
          data-for={`${index}_tooltip`}
          onClick={() => this.handleShowImage(label)}
        >
          <TableCell>
            <FiberManualRecord style={{color: backgroundColor[index]}}/>
          </TableCell>
          <TableCell>
              {buttonLabel}
          </TableCell>
          <TableCell>
            {data[index]}
          </TableCell>
        </TableRow>
      );
    });
  }

  render() {
    const {translate} = this.props;
    return (
      <Flex grow>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell colSpan='2'>{translate('review.divideAttributes.option')}</TableCell>
              <TableCell>{translate('review.count')}</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {this.renderTableBody()}
          </TableBody>
          <TableFooter>
            <TableRow>
              <TableCell colSpan='2'>{translate('review.respondents')}: {this.props.data.respondents}</TableCell>
              <TableCell>{this.props.data.answerCount}</TableCell>
            </TableRow>
          </TableFooter>
        </Table>
        <Dialog
          title="Preview"
          modal={false}
          open={this.state.showImage}
          onClose={this.handleCloseImage}
        >
          <Flex>
            <img src={this.state.image} alt={this.state.image}/>
          </Flex>
        </Dialog>
      </Flex>
    );
  }
}

LegendTable.propTypes = {
  data: PropTypes.object.isRequired,
  isOptionImage: PropTypes.bool,
};

export default localize(LegendTable, 'locale');
