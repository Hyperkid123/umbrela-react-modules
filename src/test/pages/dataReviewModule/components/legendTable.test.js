import React from 'react'
import {LegendTable} from '../../../../pages/dataReviewModule/components/legendTable';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Legend table component', () => {
  const initialProps = {};

  beforeEach(() => {
    initialProps.data = {
      answerCount: 5001,
      respondents: 3899,
      barData: {
        data: {
          labels: ['label1', 'label2'],
          datasets: [{
            backgroundColor: 'red',
            data: [5000, 1]
          }]
        }
      }
    }
    initialProps.translate = jest.fn();
    initialProps.isOptionImage = false;
  });

  it('should render correctly', () => {
    const tree = shallow(
      <LegendTable {...initialProps}/>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })

  it('should update state on show image', () => {
    const wrapper = shallow(
      <LegendTable {...initialProps} isOptionImage={true}/>
    )
    const imageUrl = 'foo.url';
    wrapper.instance().handleShowImage(imageUrl);
    expect(wrapper.state()).toEqual({showImage: true, image: imageUrl});
  });

  it('should update state on image close', () => {
    const wrapper = shallow(
      <LegendTable {...initialProps} isOptionImage={true}/>
    )
    wrapper.instance().handleCloseImage();
    expect(wrapper.state().showImage).toBeFalsy()
  })
});
