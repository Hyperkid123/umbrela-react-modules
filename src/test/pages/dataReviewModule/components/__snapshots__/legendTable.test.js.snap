// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend table component should render correctly 1`] = `
<styled.div
  grow={true}
>
  <WithStyles(Table)>
    <TableHead
      component="thead"
    >
      <WithStyles(TableRow)>
        <WithStyles(TableCell)
          colSpan="2"
        />
        <WithStyles(TableCell) />
      </WithStyles(TableRow)>
    </TableHead>
    <TableBody
      component="tbody"
    >
      <WithStyles(TableRow)
        data-for="0_tooltip"
        data-tip={true}
        key="0"
        onClick={[Function]}
      >
        <WithStyles(TableCell)>
          <pure(FiberManualRecord)
            style={
              Object {
                "color": "r",
              }
            }
          />
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          label1
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          5000
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        data-for="1_tooltip"
        data-tip={true}
        key="1"
        onClick={[Function]}
      >
        <WithStyles(TableCell)>
          <pure(FiberManualRecord)
            style={
              Object {
                "color": "e",
              }
            }
          />
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          label2
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          1
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </TableBody>
    <TableFooter
      component="tfoot"
    >
      <WithStyles(TableRow)>
        <WithStyles(TableCell)
          colSpan="2"
        >
          : 
          3899
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          5001
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </TableFooter>
  </WithStyles(Table)>
  <WithStyles(Dialog)
    modal={false}
    onClose={[Function]}
    open={false}
    title="Preview"
  >
    <styled.div>
      <img
        alt=""
        src=""
      />
    </styled.div>
  </WithStyles(Dialog)>
</styled.div>
`;
